### ISSUE TITLE
`finnish` doesn't work for functions without an explicit return type

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#57]
This will get flagged by the `finnish` rule:

```ts
function getNumber(): Observable<number> {
	return of(42);
}
```

but this will not get flagged:

```ts
function getNumber() {
	return of(42);
}
```

This is likely due to `couldReturnType` not being able to handle implicit return types.

```ts
tsTypeNode = tsNode.type ?? tsNode.body;
```

If `type` is null (no explicit return type) and `body` is a Block instead of something like a CallExpression, then the type checker will not be able to determine the type of `tsTypeNode` and so `couldReturnType` will always return false.  This is currently impacts the `finnish`, `no-finnish`, and `no-misused-observables` rules.

Implicit return types don't have test coverage in the `finnish` test suite, so this bug was never discovered (or ignored?).

**Comment by Dmytro27Ind** [comment]
I’ve also encountered this problem where implicit return types in functions aren't flagged correctly by the `finnish` rule.

As a temporary workaround, I added the following rule to enforce explicit return types:

```ts
"@typescript-eslint/explicit-function-return-type": "error"
```

This ensures that all functions have explicit return types, which indirectly resolves the issue by preventing cases where `couldReturnType` fails to handle implicit types.

While this workaround helps in the interim, it would be great to see a fix for the root cause in the `finnish` rule itself.

**Comment by JasonWeinzierl** [comment]
Agreed I also have been using `explicit-function-return-type` + `finnish` with the old library, and so I never noticed this bug until starting this fork.  I would like to get this bug fixed before v1 is released, because `no-misused-observables` now has the same problem.

**Comment by JasonWeinzierl** [comment]
This is fixed in version 0.7.2!
